_posts/2017-02-21-Rails-5-1-beta1.markdown

```diff
@@ -0,0 +1,108 @@
+
+The improvements in Rails 5.1 focus on three major parts:
+
+1. **Manage JavaScript dependencies from NPM via Yarn.** Think of [Yarn](https://yarnpkg.com/en/) like Bundler for JavaScript (it even has Yehuda Katz involved!). This makes it easy to depend on libraries like React or anything else from NPM. Everything you depend on via Yarn is then made available to be required in the asset pipeline, just like vendored dependencies would have been. Just use the binstub bin/yarn to add dependencies.
+
+2. **Optionally compile JavaScript with Webpack.** While there are a million different module bundlers/compilers for JavaScript, [Webpack](https://webpack.github.io/) is quickly emerging as the preeminent choice. We've made it easy to use Webpack with Rails through the new [Webpacker](https://github.com/rails/webpacker) gem that you can configure automatically on new projects with --webpack. This is fully compatible with the asset pipeline, which you can continue to use for images, fonts, sounds, whatever. You can even have some JavaScript on the asset pipeline and some done via Webpack. It's all managed via Yarn that's on by default.
+
+3. **Drop jQuery as a default dependency.** We used to require jQuery in order to provide features like data-remote, data-confirm, and the other parts of Rails UJS. This dependency is no longer necessary as we've rewritten rails-ujs to use vanilla JavaScript. You're of course still free to use jQuery, but you no longer have to.
+
+
+Thanks to [Liceth Ovalles](https://github.com/Liceth) for her work on Yarn integration and [Guillermo Iguaran](https://github.com/guilleiguaran) for his work on rails-ujs, Yarn, and Webpack!
+
+
+## System tests
+
+In my [2014 keynote at RailsConf](https://www.youtube.com/watch?v=9LfmrkyP81M), I spoke at length about how an overfocus on unit tests (and TDD) has lead us astray. While unit tests are part of a complete testing solution, they're not the most important one. Integration tests that verify behavior all the way from controllers through models and views should play a much bigger part. Rails already has a great answer for these baked in.
+
+But integration tests do not help you test the entire system, if that system relies on JavaScript. And most major web systems today rely at least to some extent on JavaScript. That's where system tests driven by a real browser come in.
+
+There's long been an answer for system tests like this in Ruby called [Capybara](http://teamcapybara.github.io/capybara/). It's just been kind of a journey to configure properly for Rails. So now we've [baked them straight into the framework](https://github.com/rails/rails/blob/master/actionpack/lib/action_dispatch/system_test_case.rb#L11)! You get a lovely wrapping of Capybara that's preconfigured for Chrome and enhanced to provide failure screenshots as part of Action Dispatch.
```

Comment:
Could be a good point to mention that the "journey" also used to involve DatabaseCleaner and how that's fixed too.

Ref: https://github.com/rails/rails/pull/28083